File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <iostream>
#include <vector>
using namespace std;

vector<int> vec[101];
bool visited[101] = {false};
int answer = 0;

void dfs(int item) {
    visited[item] = true;
    answer++;
    for(int it : vec[item]) {
        if(visited[it] == false) {
            dfs(it);
        }
    }
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int V, E;

    cin >> V;
    cin >> E;
    for (int i = 0; i < E; i++) {
        int start, end;
        cin >> start >> end;
        vec[start].push_back(end);
        vec[end].push_back(start);
    }

    dfs(1);

    cout << answer-1;

    return 0;
}

File: 백준/Silver/2606. 바이러스/바이러스.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const [V] = input[0].split(" ").map(Number);
const map = input.slice(2).map((_) => _.trim().split(" ").map(Number));

const edge = Array.from({ length: V + 1 }, () => []);
const visited = Array.from({ length: V + 1 }, () => false);

function dfs(item) {
  visited[item] = true;
  edge[item].forEach((v) => {
    if (!visited[v]) {
      dfs(v);
    }
  });
}

function main() {
  map.forEach(([start, end]) => {
    edge[start].push(end);
    edge[end].push(start);
  });

  dfs(1);

  console.log(visited.filter((el) => el === true).length - 1);
}

main();
